src/java/org/apache/cassandra/net/FrameDecoderLegacy.java

```diff
@@ -1,184 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.cassandra.net;
-
-import java.nio.ByteBuffer;
-import java.util.Collection;
-
-import io.netty.channel.ChannelPipeline;
-
-import static java.lang.Math.max;
-import static org.apache.cassandra.net.OutboundConnections.LARGE_MESSAGE_THRESHOLD;
-
-/**
- * {@link InboundMessageHandler} operates on frames that adhere to a certain contract
- * (see {@link FrameDecoder.IntactFrame} and {@link FrameDecoder.CorruptFrame} javadoc).
- *
- * Legacy (pre-4.0) messaging protocol does not natively support framing, however. The job
- * of {@link FrameDecoderLegacy} is turn a raw stream of messages, serialized back to back,
- * into a sequence of frames that adhere to 4.0+ conventions.
- */
-class FrameDecoderLegacy extends FrameDecoder
```

Comment:
The removal of the legacy frame decoders leaves some obsolete comments  in the class JavaDoc for `FrameDecoder`: https://github.com/apache/cassandra/blob/530750f780b1b4a683263337574f2fa5cabd4dce/src/java/org/apache/cassandra/net/FrameDecoder.java#L54-L57

Reply:
One of the `if` branches in `FrameDecoder#channelRead(ChannelHandlerContext, Object)` also seems related with the disappeared legacy decoders: https://github.com/apache/cassandra/blob/530750f780b1b4a683263337574f2fa5cabd4dce/src/java/org/apache/cassandra/net/FrameDecoder.java#L279-L282